### PR TITLE
SDIT-873 Skeleton POST for creating an adjudication

### PR DESCRIPTION
**Pull request by andymarke**
TODO

- Implementation
- Tests around the actual POST
- Better DSL for staff accounts

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt [hunk 1]
```diff
@@ -104,7 +107,12 @@ class AdjudicationResource(
       ApiResponse(
         responseCode = "200",
         description = "Adjudication Information Returned",
-        content = [Content(mediaType = "application/json", schema = Schema(implementation = AdjudicationResponse::class))],
+        content = [
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt:110]: just formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt [hunk 2]
```diff
@@ -144,6 +152,72 @@ class AdjudicationResource(
...
+        description = "Prisoner does not exist",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "409",
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt:203]: not sure on this ....

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsResourceIntTest.kt [hunk 3]
```diff
@@ -714,4 +717,104 @@ class AdjudicationsResourceIntTest : IntegrationTestBase() {
...
+    private lateinit var prisoner: Offender
+    private lateinit var reportingStaff: Staff
+    private lateinit var reportingStaffAccount: StaffUserAccount
+
+    @BeforeEach
+    fun createPrisoner() {
+      nomisDataBuilder.build {
+        prisoner = offender(nomsId = "A1965NM") { booking { } }
+        reportingStaff = staff {}
+        reportingStaffAccount = staffAccount("JANESTAFF", staff = reportingStaff)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsResourceIntTest.kt:733]: I will improve this - should be nested under `staff`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt:203]: Can discuss

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**